enas/cifar10/models.py

```diff
@@ -85,18 +85,22 @@ def __init__(self,
       self.lr_dec_every = lr_dec_every * self.num_train_batches
 
       def _pre_process(x):
```

Comment:
preprocess will probably be worth moving to outside this class so we can use datasets that don't fit into memory. When I mentioned input tensors, a simple example API that uses them is the keras [mnist tfrecord example](https://github.com/keras-team/keras/blob/master/examples/mnist_tfrecord.py),  one that is tf only is the [tf dataset API](https://www.tensorflow.org/api_docs/python/tf/data/Dataset). The other API we use in our project is a keras/python generator, which you can see in most of the keras examples folder.